### PR TITLE
Implement JNI wrapper

### PR DESCRIPTION
**Pull request by minmingzhu**
1. add JNI wrapper
2. modify compiler config
3. add unit test

Signed-off-by: minmingzhu <minming.zhu@intel.com>

## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## Does this PR also require the following changes?

- CI
- Documentation
- Example

### CODE REVIEW

**Review by xwu99 — CHANGES_REQUESTED** [review]
I didn't finish review yet, but you can change the code from current comments.

File: mllib-dal/src/main/java/com/intel/oneapi/dal/table/HomogenTableTemplate.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.intel.oneapi.dal.table;
+
+public interface HomogenTableTemplate extends HomogenTableIface{
```
[the remark this answers is not in the document]
Comment (minmingzhu) [reply mllib-dal/src/main/java/com/intel/oneapi/dal/table/HomogenTableTemplate.java:3]: I refer to oneapi/dal code that homogen_table_impl extends homogen_table_template. You suggest we can move getPullRowsIface and  getPullColumnIface to TableIface/HomogenTableIface and remove HomogenTableTemplate interface?

[the remark this answers is not in the document]
Comment (xwu99) [reply mllib-dal/src/main/java/com/intel/oneapi/dal/table/HomogenTableTemplate.java:3]: > I refer to oneapi/dal code that homogen_table_impl extends homogen_table_template. You suggest we can move getPullRowsIface and getPullColumnIface to TableIface or HomogenTableIface and remove HomogenTableTemplate interface?

This class is unnecessary. We don't need to follow the previous interfaces. We can defined interfaces to serve our own needs. I don't like too many level inheritance, If we need more abstraction, we can just refactor in the future.

**Comment by xwu99** [comment]
I dont think your tests are running in CI. You may need to remove -Dtest=none from mvn command line.

**Pull request merged**